### PR TITLE
Improve error display

### LINKED ISSUE #257: Add loading animation to the "Debit or Credit Card" button on checkout
On checkout, after clicking the "Debit or Credit Card" button, it takes about 3 seconds for the form to open.

I've already confirmed with your support that it's a known issue. I understand there are limitations regarding the integration with PayPal.

I suggest asking PayPal to add a processing/loading animation to the button after clicking it. This way, the user gets visual feedback assuring everything is okay and that they should wait.

### LINKED ISSUE #457: Display API response errors in checkout page with user-friendly error message (257)
When PayPal rejects a payment, an error like this will be displayed in the checkout:
```UNPROCESSABLE_ENTITY The requested action could not be performed, semantically incorrect, or failed business validation. https://developer.paypal.com/docs/api/orders/v2/#error-PAYEE_ACCOUNT_RESTRICTED```

Suggestion: Replace it with an error like: PayPal refused the transaction. PAYEE_ACCOUNT_RESTRICTED: The merchant account is restricted.

Orders API errors can be found here: https://developer.paypal.com/api/rest/reference/orders/v2/errors/ functionality to swap out the displayed error with the description in the checkout 

The most frequent errors that could use better explanation:
- INSTRUMENT_DECLINED
- TRANSACTION_REFUSED
- DUPLICATE_INVOICE_ID

### Negative testing payee account restricted plugin
[negative-testing-payee-account-resticted.php.zip](https://github.com/woocommerce/woocommerce-paypal-payments/files/7970281/negative-testing-payee-account-resticted.php.zip)

### To Reproduce
1. Install and activate 'Negative testing payee account restricted' plugin attached above.
2. Place an order.
3. Error message is displayed.

### Expected Behavior

`PAYEE_ACCOUNT_RESTRICTED The merchant account is restricted.` error displayed in checkout page.

### Actual Behavior

Generic error displayed in checkout page.

### Environment
 - WordPress Version: 5.9
 - WooCommerce Version: 6.1
 - Plugin Version: 1.6.4

### PR DESCRIPTION
**Pull request by Dinamiko**
Fixes #457 and #456.

### CODE REVIEW

File: modules/ppcp-button/resources/js/modules/ActionHandler/CheckoutActionHandler.js [hunk 1]
```diff
@@ -48,7 +48,11 @@ class CheckoutActionHandler {
                                 .querySelector('ul')
                         );
                     } else {
-                        errorHandler.message(data.data.message, true);
+                        if (data.data.details.length > 0) {
+                            errorHandler.message(data.data.details[0].issue + ' ' + data.data.details[0].description, true);
```
Comment (AlexP11223) [inline modules/ppcp-button/resources/js/modules/ActionHandler/CheckoutActionHandler.js]: should we display only the first one? Maybe better e.g. 

    data.data.details.map(d => `${d.issue} ${d.description}`).join('<br/>')

Comment (Dinamiko) [reply modules/ppcp-button/resources/js/modules/ActionHandler/CheckoutActionHandler.js]: @AlexP11223 Done, thanks!

File: modules/ppcp-wc-gateway/src/Gateway/ProcessPaymentTrait.php [hunk 2]
```diff
@@ -298,6 +298,12 @@ function ( Authorization $authorization ): bool {
 				);
 			}
 
+			$error_message = $error->getMessage();
+			if ( $error->issues() ) {
+				$error_message = $error->issues()[0]->issue . ' ' . $error->issues()[0]->description;
```
Comment (AlexP11223) [inline modules/ppcp-wc-gateway/src/Gateway/ProcessPaymentTrait.php]: and in other places (in PHP, ...) too then. `implode(array_map(.....))`

Comment (Dinamiko) [reply modules/ppcp-wc-gateway/src/Gateway/ProcessPaymentTrait.php]: Done! :)

**Review by AlexP11223 — APPROVED** [review]

**Pull request merged**